### PR TITLE
Add new test to verify that the listener receives the correct path an…

### PR DESCRIPTION
**Pull request by bainian12345**

### CODE REVIEW

File: src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java [hunk 1]
```diff
@@ -191,6 +192,55 @@ public void requestHandlerTest() throws IOException {
...
+		listener.setRequestHandler((context) -> {
+			try {
+				assertNotNull("Listener should have received a valid http context.", context);
+				assertNotNull("Listener should have received a valid http request.", context.getRequest());
+				
+				URI requestUri = context.getRequest().getUri();
+				assertNotNull("Listener should have a URI from request", requestUri);
+				assertEquals(
+						"The path wasn't the expected value", 
+						"/bailiu-relay-hc/bailiu-relay-hc" + extraPath,
```
Comment (dlstucki) [inline src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java]: >bailiu-relay-hc/bailiu-relay-hc [](start = 8, length = 31)

This won't work for other people, right?

File: src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java [hunk 2]
```diff
@@ -191,6 +192,55 @@ public void requestHandlerTest() throws IOException {
...
+				assertEquals(
+						"The path wasn't the expected value", 
+						"/bailiu-relay-hc/bailiu-relay-hc" + extraPath, 
+						requestUri.getPath()
+				);
+				assertEquals("The query wasn't the expected value", extraQuery, requestUri.getQuery());
+				
+				requestReceived.complete(null);
+			} catch (Throwable ex) {
+				requestReceived.completeExceptionally(ex);
```
Comment (dlstucki) [inline src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java]: I love this. At first I wondered how asserts in this handler would get reported.

File: src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java (hunk 1 above)
Comment (dlstucki) [reply src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java]: I think it should be based on TestUtil.ENTITY_PATH

---
In reply to: [263873685](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263873685) [](ancestors = 263873685)

Comment (bainian12345) [reply src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java]: Yes it should be TestUtil.ENTITY_PATH, although I find it weird why it's appended again.

---
In reply to: [263873685](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263873685) [](ancestors = 263873685)

Comment (dlstucki) [reply src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java]: Is that a bug?

---
In reply to: [263976580](https://github.com/Azure/azure-relay-java/pull/31#discussion_r263976580) [](ancestors = 263976580,263873685)

**Pull request merged**